Payload_Type/Apollo/agent_code/Apollo/C2Profiles/DefaultProfile.cs

```diff
@@ -473,13 +487,19 @@ override public Mythic.Structs.TaskQueue GetMessages(Apollo.Agent agent)
                 {
                     response.SocksDatagrams = resp.socks;
                 }
+                if (resp.rportfwds != null)
+                {
+                    response.PortFwdDg = new PortFwdDatagram[1];
+                    response.PortFwdDg[0].data = JsonConvert.DeserializeObject<Dictionary<String, Dictionary<String, Dictionary<String, Dictionary<String, Dictionary<int,String>>>>>>(resp.rportfwds[0]);
```

Comment:
This datagram should be changed to a struct for code maintainability/readability. Five nested dictionaries, especially in subsequent function calls, starts to make it difficult to know how things are going wrong.